test/unit/scanner/index.test.ts

```diff
@@ -0,0 +1,58 @@
+import * as tap from 'tap';
+
+import * as scanner from '../../../src/scanner';
+import { IWorkload } from '../../../src/transmitter/types';
+
+tap.test('getUniqueImages()', async (t) => {
+    const workload: Partial<IWorkload>[] = [
+        // 1.DCR
+        {
+            imageName: 'redis:latest',
+            imageId: 'docker.io/library/redis@sha256:8e9f8546050da8aae393a41d65ad37166b4f0d8131d627a520c0f0451742e9d6'
+        },
+        // 2.Duplicate to verify uniqueness
+        {
+            imageName: 'redis:latest',
+            imageId: 'docker.io/library/redis@sha256:8e9f8546050da8aae393a41d65ad37166b4f0d8131d627a520c0f0451742e9d6'
+        },
+        // 3. With SHA instead of tag
+        {
+            imageName: 'redis@sha256:8e9f8546050da8aae393a41d65ad37166b4f0d8131d627a520c0f0451742e9d6',
+            imageId: 'docker.io/library/redis@sha256:8e9f8546050da8aae393a41d65ad37166b4f0d8131d627a520c0f0451742e9d6'
+        },
+        // 4. With SHA missing in imageId
+        {
+            imageName: 'redis:prod',
+            imageId: 'docker.io/library/redis:eaa6f054e4a140bc3a1696cc7b1e84529e7e9567',
+        },
+        // 5. GCR
+        {
+            imageName: 'gcr.io/test-dummy/redis:latest',
+            imageId: 'sha256:8e9f8546050da8aae393a41d65ad37166b4f0d8131d627a520c0f0451742e9d6'
+        },
+        // 6. ECR
+        {
+            imageName: '291964488713.dkr.ecr.us-east-2.amazonaws.com/snyk/redis:latest',
+            imageId: 'sha256:8e9f8546050da8aae393a41d65ad37166b4f0d8131d627a520c0f0451742e9d6'
+        },
+    ];
```

Comment:
I think we need cases for the following:

```typescript
imageId: 'docker://sha256:abcd',
imageId: 'docker://name@sha256:abcd',
imageId: 'docker-pullable://name@sha256:abcd',
```

Reply:
Cool. Will add test cases!

Reply:
Still, the first case is not supported for now. So I would skip it. When we get some sample data from the logger, I might add this. WDYT?

Reply:
Sure thing, as long as we can fallback to the old way of pulling by name/tag 🙏